File: number3.cpp
#include <iostream>
#include <cmath>
int row(int first, int second);
bool input(int& first, int& second);
int gcd(int64_t idealNum, int64_t idealDin);
int main(){
	int first, second;
	if(input(first,second))
		row(first, second);
	else
		std::cout<<"Введите корректные значения"<<std::endl;
	return 0;
}
bool input(int& first, int& second){
	bool isDone=1;
	if(!(std::cin>>first && first>0 && first<11 && std::cin>>second && second>0 && second<11))
		isDone=0;
	return isDone;
}
int row(int first, int second){
	long double sum=0,idealError=1e14;
	if(second==1){
		std::cout<<"infinity"<<std::endl;
		return 0;
	}
	for(int n=1;n<200;++n){
		sum+=std::pow(n,first)/std::pow(second,n);
	}
	int64_t idealNum=0, idealDen=1;
	while(idealDen<10000){
		idealNum=std::round(sum*idealDen);
		long double error=std::abs(sum-(long double)idealNum/idealDen);	
		if(error<idealError){
			idealError=error;
			first=idealNum;
			second=idealDen;
		}
		++idealDen;
	}
	int del=gcd(first, second);
	std::cout<<first/del << "/" <<second/del <<std::endl;
	return 0;
}
int gcd(int64_t idealNum, int64_t idealDen){
	while(idealDen!=0){
		idealNum%=idealDen;
		std::swap(idealNum,idealDen);
	}
	return idealNum;
}

File: number3.go
package main

import(
    "fmt"
    "math")

    func input(first, second *int) bool {
    _, err : = fmt.Scan(first, second) if err != nil { return false }
    return *first > 0 && *first < 11 && *second > 0 && *second < 11
}

func gcd(idealNum, idealDen int64) int64 {
        for
            idealDen != 0 { idealNum %= idealDen idealNum, idealDen = idealDen, idealNum }
        return idealNum
}

func row(first, second int) int {
    if second == 1 {
		fmt.Println("infinity")
		return 0
	}

	var sum float64 = 0.0
	for n := 1;
    n < 200; n++ {
		sum += math.Pow(float64(n), float64(first)) / math.Pow(float64(second), float64(n))
	}

	var idealNum, idealDen int64 = 0, 1
	idealError := 1e14

	for den := int64(1);
    den < 10000;
    den++ {
    num:
        = int64(math.Round(sum * float64(den))) err
            : = math.Abs(sum - float64(num) / float64(den)) if err < idealError {
            idealError = err idealNum = num idealDen = den
        }
    }

div:
    = gcd(idealNum, idealDen) fmt.Printf("%d/%d\n", idealNum / div, idealDen / div) return 0
}

func main() {
    var first, second int if input (&first, &second) { row(first, second) }
    else {
        fmt.Println("Введите корректные значения")
    }
}
